internal/resources/utils/clog/clog.go

```diff
@@ -0,0 +1,71 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package clog
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type Logger struct {
+	*logp.Logger
+}
+
+func (l *Logger) Errorf(template string, args ...any) {
+	// Downgrade context.Canceled errors to warning level
+	if hasErrorType(context.Canceled, args...) {
+		l.Warnf(template, args...)
+		return
+	}
```

Comment:
this is reason for this PR - downgrading errors to warnings when we see `context canceled`